Products/ZenEvents/zentrap.py

```diff
@@ -590,11 +643,18 @@ def _add_varbind_detail(self, result, oid, value):
         result[detail_name].append(str(value))
 
     def decodeSnmpv1(self, addr, pdu):
+
         result = {"snmpVersion": "1"}
+        result["device"] = addr[0]
 
         variables = self.getResult(pdu)
 
-        result["device"] = addr[0]
+        self.log.debug("SNMPv1 pdu has agent_addr: %s",
+                       str(hasattr(pdu, 'agent_addr')))
+
+        if hasattr(pdu, 'agent_addr'):
+            origin = '.'.join([str(i) for i in pdu.agent_addr])
```

Comment:
List comprehension is unnecessary here.  A generator comprehension will work just fine.  E.g.
```
    origin = '.'.join(str(i) for i in pdu.agent_addr)
```